os/pm/pm_idle.c

```diff
@@ -0,0 +1,135 @@
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * os/pm/pm_idle.c
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <debug.h>
+
+#include <tinyara/board.h>
+#include <tinyara/pm/pm.h>
+
+#include <tinyara/irq.h>
+
+/****************************************************************************
+ * Pre-processor Definitions
+ ****************************************************************************/
+
+#define PM_IDLE_DOMAIN	0
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+/****************************************************************************
+ * Name: pm_idle_process
+ *
+ * Description:
+ *   Perform IDLE state power management.
+ *
+ ****************************************************************************/
+static void pm_idle_process(void)
+{
+	static int oldstate = PM_NORMAL;
+	int newstate;
+	irqstate_t flags;
+	int ret;
+
+	/* Decide, which power saving level can be obtained */
+	newstate = pm_checkstate(PM_IDLE_DOMAIN);
+
+	/* Check for state changes */
+	if (newstate != oldstate) {
+		flags = irqsave();
+
+		/* Then force the global state change */
+		ret = pm_changestate(PM_IDLE_DOMAIN, newstate);
+		if (ret < 0) {
+			/* The new state change failed, revert to the preceding state */
+			pm_changestate(PM_IDLE_DOMAIN, oldstate);
+		} else {
+			/* SOC-specific power management logic */
+			ret = up_pm_switch_state(newstate);
+			if (ret < 0) {
+				/* Staying in same state as SOC returned error */
+				ret = up_pm_switch_state(oldstate);
+				ret = pm_changestate(PM_IDLE_DOMAIN, oldstate);
+			} else {
+				/* Save the new state */
+				oldstate = newstate;
+			}
+		}
+
+		irqrestore(flags);
+	}
+}
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: pm_process_idle
+ *
+ * Description:
+ *   pm_process_idle() is the logic that will be executed when their is no other
+ *   ready-to-run task.  This is processor idle time and will continue until
+ *   some interrupt occurs to cause a context switch from the idle task.
+ *
+ ****************************************************************************/
+void pm_process_idle(void)
```

Comment:
pm_process_idle calls pm_idle_process? Why? There is no additional operation.

Reply:
will merge them into one API